sdk/resourcemanagerhybrid/azure-resourcemanager/src/main/java/com/azure/resourcemanager/AzureResourceManager.java

```diff
@@ -313,6 +324,59 @@ public AzureResourceManager withDefaultSubscription() {
         }
     }
 
+    /**
+     * Gets AzureEnvironment from ARM endpoint.
+     *
+     * @param armEndpoint the URL of ARM endpoint.
+     * @param httpClient the HTTP client for requests.
+     * @return the AzureEnvironment
+     * @throws HttpResponseException thrown if the request is rejected by server, or failed to parse metadata.
+     * @throws RuntimeException thrown if ARM endpoint is invalid, or other errors.
+     */
+    public static AzureEnvironment getAzureEnvironmentFromArmEndpoint(String armEndpoint, HttpClient httpClient) {
```

Comment:
This is likely useful for the Azure Resourcemanager package as well (not just for hybrid). you may want to make another PR later once we validate..

Reply:
Agree.

One difference is the number of metadata items. While hybrid appears only have 1 item in metadata, `https://management.azure.com` have multiple metadata distinguished by name, e.g. `AzureCloud`.

For for it, we probably have this method, and another overloaded method with an additional parameter of `name` to choose from the multiple metadata items.

Reply:
Yes. You are correct. Azurestack metadata will always have information about only one cloud. 

Azure endpoints will return all the public clouds metadata information